lib/libzfs/libzfs_util.c

```diff
@@ -596,27 +596,47 @@ zfs_strdup(libzfs_handle_t *hdl, const char *str)
  * Convert a number to an appropriately human-readable output.
  */
 void
-zfs_nicenum(uint64_t num, char *buf, size_t buflen)
+__zfs_nicenum(uint64_t num, char *buf, size_t buflen,
```

Comment:
I don't have a strong opinion of the head fake, but note that you've taken a 40 LOC function that fits nicely on one screen and grown it to 70 LOC, not counting the head fakes. Maybe just a second 40 LOC is easier to read?

NB, there are other places where zfs_nicenum is used to display things that should be base10, but that can be tackled as a separate PR